server/internal/server/completions.go

```diff
@@ -57,22 +57,24 @@ func (s *S) CreateChatCompletion(
 		return
 	}
 
-	dest, err := s.engineGetter.GetEngineForModel(req.Context(), createReq.Model)
-	if err != nil {
-		http.Error(w, fmt.Sprintf("Failed to find pod to route the request: %s", err), http.StatusInternalServerError)
-		return
-	}
-
 	// Check if the specified model is available
-	ctx := auth.CarryMetadataFromHTTPHeader(req.Context(), req.Header)
-	if _, err := s.modelClient.GetModel(ctx, &mv1.GetModelRequest{
-		Id: createReq.Model,
-	}); err != nil {
-		if status.Code(err) == codes.NotFound {
-			http.Error(w, fmt.Sprintf("Model not found: %s", createReq.Model), http.StatusBadRequest)
+	if s.enableAuth {
```

Comment:
Hmm, this logic is not particularly tied to authentication? Is this related to your dev setting to avoid dep to model-manager-server?

Reply:
I think we can merge, but I'm curious your dev setting. I was trying to see what's the best way to test inference-manager without less dependency.

Reply:
My understanding was: `GetModel` fails if 

- model does not exist at all. `GetEngineForModel` returns error for this case too, so by-pass the check if `Auth` is disabled, no need to check twice?
- model exists but is not accessible for this user. `GetEngineForModel` does not check for this.

Not sure if any other cases missed?

Reply:
Yeah, that's correct. I wasn't sure whether we need to have a special logic here. It seems to me that we need to this if we disable auth in `inference-manager-server`, but keeps the auth in `model-manager-server`?

Reply:
Again we can merge the PR, but I was just curious..